xmscore/python/misc/PyUtils.cpp

```diff
@@ -93,6 +125,41 @@ py::iterable PyIterFromVecPt3d(const VecPt3d& pts)
   return a;
 } // PyIterFromVecPt3d
 //------------------------------------------------------------------------------
+/// \brief Create VecPt2d from py::iterable
+/// \param[in] pt: py::iterable object that represents a VecPt2d
+/// \return a VecPt2d
+//------------------------------------------------------------------------------
+boost::shared_ptr<VecPt2d> VecPt2dFromPyIter(const py::iterable& pts)
+{
+  boost::shared_ptr<xms::VecPt2d> vec_pts(new xms::VecPt2d());
+  for (auto item : pts) {
+    if(!py::isinstance<py::iterable>(item)) {
+      throw py::type_error("First arg must be a n-tuple of 0, 1, or 2-tuples");
```

Comment:
Same as the last item. We should make sure the check and the error message match